### ISSUE TITLE
PHP Fatal error:  Call to undefined method WP_Error::get() in 0.9.7, 0.9.8

### ISSUE THREAD

**Issue by aaemnnosttv** [deliciousbrains/wp-amazon-s3-and-cloudfront#292]
I got this error after upgrading to 0.9.7, as well as 0.9.8 (which is the current stable release).

```
PHP Fatal error:  Call to undefined method WP_Error::get() in .../app/plugins/amazon-s3-and-cloudfront/classes/amazon-s3-and-cloudfront.php on line 1822
PHP Stack trace:
PHP   1. {main}() .../wp/wp-admin/post.php:0
PHP   2. include() .../wp/wp-admin/post.php:218
PHP   3. do_meta_boxes() .../wp/wp-admin/edit-form-advanced.php:620
PHP   4. call_user_func:{.../wp/wp-admin/includes/template.php:1090}() .../wp/wp-admin/includes/template.php:1090
PHP   5. post_thumbnail_meta_box() .../wp/wp-admin/includes/template.php:1090
PHP   6. _wp_post_thumbnail_html() .../wp/wp-admin/includes/meta-boxes.php:1117
PHP   7. wp_get_attachment_image() .../wp/wp-admin/includes/post.php:1380
PHP   8. wp_get_attachment_image_src() .../wp/wp-includes/media.php:746
PHP   9. image_downsize() .../wp/wp-includes/media.php:697
PHP  10. wp_attachment_is_image() .../wp/wp-includes/media.php:155
PHP  11. wp_attachment_is() .../wp/wp-includes/post.php:5222
PHP  12. get_attached_file() .../wp/wp-includes/post.php:5176
PHP  13. apply_filters() .../wp/wp-includes/post.php:202
PHP  14. call_user_func_array:{.../wp/wp-includes/plugin.php:213}() .../wp/wp-includes/plugin.php:213
PHP  15. Amazon_S3_And_CloudFront->get_attached_file() .../wp/wp-includes/plugin.php:213
PHP  16. apply_filters() .../app/plugins/amazon-s3-and-cloudfront/classes/amazon-s3-and-cloudfront.php:1537
PHP  17. call_user_func_array:{.../wp/wp-includes/plugin.php:213}() .../wp/wp-includes/plugin.php:213
PHP  18. AS3CF_Plugin_Compatibility->get_stream_wrapper_file() .../wp/wp-includes/plugin.php:213
PHP  19. AS3CF_Plugin_Compatibility->register_stream_wrapper() .../app/plugins/amazon-s3-and-cloudfront/classes/as3cf-plugin-compatibility.php:685
PHP  20. Amazon_S3_And_CloudFront->get_s3client() .../app/plugins/amazon-s3-and-cloudfront/classes/as3cf-plugin-compatibility.php:629
```

**Comment by polevaultweb** [comment]
@aaemnnosttv how are you entering your API key and secret? In the UI or with the constants? Could you post your diagnostic info from the support tab in the settings so I can try to recreate please.

**Comment by aaemnnosttv** [comment]
I define them with the constants which are set using Dotenv:
```
define('AWS_ACCESS_KEY_ID', getenv('AWS_ACCESS_KEY_ID'));
define('AWS_SECRET_ACCESS_KEY', getenv('AWS_SECRET_ACCESS_KEY'));
define('AS3CF_BUCKET', getenv('AS3CF_BUCKET'));
```

This is on my local environment where I don't want to be uploading stuff to S3, so I never defined those in my `.env`.  So those are all defined as `false`.  I still want the url rewriting though, so I need to keep the plugin activated.

Like I said, it wasn't a problem for me in 0.9.6 or before, so it shouldn't be blowing up just because those are empty.

**Comment by polevaultweb** [comment]
@aaemnnosttv thanks for that further info. You are right it shouldn't be blowing up like that!

**Comment by A5hleyRich** [comment]
Resolved in version 0.9.9.
